src/otx/v2/api/core/engine.py

```diff
@@ -23,15 +24,17 @@ class Engine:
     )
     """
 
-    def __init__(self, work_dir: str | Path | None) -> None:
+    def __init__(self, task: TaskType, work_dir: str | Path | None = None) -> None:
```

Comment:
Personally, I am good to have `task` when creating engine. But, I just want to know the consensus behind. Do we have any agreement from others?

Reply:
We don't have agreement of this and I also cautious to add parameter to global api. However, I believe engine should have task attribute if our design is a single engine across all tasks.  
How about discuss about this in this PR or execution forum?

Reply:
If I remember correctly, we had already agreed to use the MMXEngine as one (If we clean up our API of Exporting & Prediction later), and this could be part of that process.
For now, while the actual Engine usage example doesn't change, but it would be nice to have a discussion in the future about whether we should always include a task when creating engine.